core/src/smartcontracts/isi/account.rs

```diff
@@ -201,24 +201,25 @@ pub mod isi {
 
             let wsv_clone = wsv.clone();
 
-            wsv_clone.permission_token_definitions()
+            wsv_clone
+                .permission_token_definitions()
                 .get(permission.definition_id())
                 .ok_or_else(|| {
                     FindError::PermissionTokenDefinition(permission.definition_id().clone())
                 })?;
 
-            let temporary_error = wsv.modify_account(&account_id, |account| {
+            wsv.modify_account(&account_id, |account| {
                 let id = account.id();
                 if wsv_clone.account_contains_inherent_permission(id, &permission) {
                     return Err(ValidationError::new("Permission already exists").into());
                 }
 
                 Ok(AccountEvent::PermissionAdded(id.clone()))
-            });
-            if temporary_error.is_ok() {
+            })
+            .and_then(|_| {
                 wsv.add_account_permission(&account_id, permission);
-            }
-            temporary_error
+                Ok(())
+            })
```

Comment:
I think [`map`](https://doc.rust-lang.org/core/result/enum.Result.html#method.map) is what you was looking for

Reply:
good suggestion